### PR TITLE
Implement Admin#advertiser_signup

### PR DESCRIPTION
**Pull request by RKushnir**

### CODE REVIEW

File: lib/soapy_cake/response.rb [hunk 1]
```diff
@@ -79,12 +79,14 @@ def error_check_special_case?
...
       return true if sax.for_tag(:MediaType).count > 0
+      return true if sax.for_tag(:AdvertiserResult).first == 'Advertiser Added Successfully'
 
       false
     end
 
     def error_message
-      @error_message ||= sax.for_tag(:message).first || sax.for_tag(:Text).first || 'Unknown error'
+      @error_message ||= sax.for_tag(:message).first || sax.for_tag(:Text).first ||
+        sax.for_tag(:AdvertiserResult).first || 'Unknown error'
```
Comment (dziemba) [inline lib/soapy_cake/response.rb]: Do we really need this if we already have the change in line 82?

Comment (RKushnir) [inline lib/soapy_cake/response.rb]: To get the exact error message, yes. Otherwise for advertiser signup errors it will just default to 'Unknown error'.

Comment (RKushnir) [inline lib/soapy_cake/response.rb]: Example of such error:
```xml
<AdvertiserResult>Missing Fields: Street, City, State, Zip Code, Country, Work Phone</AdvertiserResult>
```

Comment (dziemba) [inline lib/soapy_cake/response.rb]: Hmm...

**Comment by dziemba** [comment]
What do you think of using `ADMIN ADDEDIT Advertisers` instead of the `SIGNUP` API? That way we don't have to deal with the shitty error handling...

**Pull request closed without merging**